pkg/dotnet/core_deps/parse.go

```diff
@@ -31,30 +50,60 @@ func (p *Parser) Parse(r dio.ReadSeekerAt) ([]types.Library, []types.Dependency,
 	}
 
 	var libraries []types.Library
-	for nameVer, lib := range depsFile.Libraries {
-		if !strings.EqualFold(lib.Type, "package") {
+	var deps []types.Dependency
+	for pkgNameVersion, target := range depsFile.Targets[depsFile.RuntimeTarget.Name] {
+		library, ok := depsFile.Libraries[pkgNameVersion]
+		if !ok {
 			continue
 		}
 
-		split := strings.Split(nameVer, "/")
-		if len(split) != 2 {
-			// Invalid name
-			log.Logger.Warnf("Cannot parse .NET library version from: %s", nameVer)
+		name, version := splitNameVer(pkgNameVersion)
+		if name == "" || version == "" {
 			continue
 		}
 
-		libraries = append(libraries, types.Library{
-			Name:      split[0],
-			Version:   split[1],
-			Locations: []types.Location{{StartLine: lib.StartLine, EndLine: lib.EndLine}},
+		lib := types.Library{
+			ID:        utils.PackageID(name, version),
+			Name:      name,
+			Version:   version,
+			Locations: []types.Location{{StartLine: library.StartLine, EndLine: library.EndLine}},
+		}
+
+		var childDeps []string
+		for depName, depVersion := range target.Dependencies {
+			nameVer := joinNameVer(depName, depVersion)
+			_, ok := depsFile.Libraries[nameVer]
+			if !ok {
+				continue
+			}
+			depID := utils.PackageID(depName, depVersion)
+			childDeps = append(childDeps, depID)
+		}
+
+		deps = append(deps, types.Dependency{
+			ID:        lib.ID,
+			DependsOn: childDeps,
 		})
```

Comment:
saving deps without child deps makes no sense.
```suggestion
		if len(childDeps) > 0{
			deps = append(deps, types.Dependency{
				ID:        lib.ID,
				DependsOn: childDeps,
			})
		}
```